runelite-client/src/main/java/net/runelite/client/plugins/zalcano/ZalcanoPlugin.java

```diff
@@ -111,6 +135,11 @@ public void onGameStateChanged(GameStateChanged event)
 		{
 			rocks.clear();
 		}
+		else
+		{
+			inCavern = checkArea();
```

Comment:
This and the rocks.clear() should just be in `if (event.getGameState() == GameState.LOADING)`